src/pudl/settings.py

```diff
@@ -27,35 +27,32 @@ class GenericDatasetSettings(BaseModel, abc.ABC):
     """
     An abstract pydantic model for generic datasets.
 
-    Each dataset must specify working tables are partitions.
-    A dataset can have an arbitrary number of partitioins.
+    Each dataset must specify working tables and partitions.
+    A dataset can have an arbitrary number of partitions.
     """
 
-    tables: List
-
-    @property
-    @abc.abstractmethod
-    def working_partitions(cls) -> Dict[str, List[Any]]:  # noqa: N805
-        """Abstract working_partitions property."""
-        return cls.working_partitions
-
-    @property
-    @abc.abstractmethod
-    def working_tables(cls) -> List:  # noqa: N805
-        """Abstract working_tables property."""
-        return cls.working_tables
+    tables: List[str]
+    data_source: ClassVar[DataSource]
```

Comment:
I added type annotations here. Is there any reason why we wouldn't want to do that? Should the `DataSource` type also be listed in the classes that inherit from this one? Or does it only need to be here?

Reply:
I think we can actually just remove `data_source` and add type annotations in the subclasses. We also don't need this to be an abstract base class anymore, which I meant to change and forgot. I can push those changes if that sounds reasonable